packages/kbn-search-api-panels/components/ingest_pipelines/ingest_pipeline_panel.tsx

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useState, useMemo } from 'react';
+
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiSpacer,
+  EuiText,
+  EuiTitle,
+  EuiBadge,
+  EuiSuperSelect,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { IngestGetPipelineResponse } from '@elastic/elasticsearch/lib/api/types';
+import { createIngestPipelineOptions } from './ingest_pipeline_options';
+
+interface IngestPipelinePanelProps {
+  setSelectedPipeline: (pipeline: string) => void;
+  ingestPipelinesData?: IngestGetPipelineResponse;
+  defaultIngestPipeline: string;
+}
+
+export const IngestPipelinePanel: React.FC<IngestPipelinePanelProps> = ({
+  setSelectedPipeline,
+  ingestPipelinesData,
+  defaultIngestPipeline,
+}) => {
+  const options = useMemo(
+    () => createIngestPipelineOptions(ingestPipelinesData, defaultIngestPipeline),
+    [ingestPipelinesData, defaultIngestPipeline]
+  );
+
+  const [selected, setSelected] = useState<string>();
```

Comment:
we're managing this state in 2 places here and in the overview. If we're accepting `setSelectedPipeline` as a prop we should also send the selected pipeline as a prop too. These two state could get out of sync in `setSelectedPipeline` is called from another place.

Reply:
Updated